armi/physics/neutronics/globalFlux/globalFluxInterface.py

```diff
@@ -946,7 +942,19 @@ def updateFluenceAndDpa(self, stepTimeInSeconds, blockList=None):
 
             # also set the burnup peaking. Requires burnup to be up-to-date
             # (this should run AFTER burnup has been updated)
-            b.p.percentBuPeak = b.p.percentBu * burnupPeakingFactor
+            # b.p.percentBu is expected to have been updated elsewhere
+            peakRate = None
+            if b.p.buRatePeak:
+                # best case scenario, we have peak burnup rate
+                peakRate = b.p.percentBuPeak
+            elif b.p.buRate:
+                # use whatever peaking factor we can find
+                peakRate = b.p.buRate * self.getBurnupPeakingFactor(b)
+            if peakRate:
+                b.p.percentBuPeak = b.p.percentBuPeak + peakRate * stepTimeInSeconds
+            else:
+                # bad assumption.... assumes peaking is same at each position through shuffling/irradiation history...
+                b.p.percentBuPeak = b.p.percentBu * self.getBurnupPeakingFactor(b)
```

Comment:
If we acknowlege this to be a bad assumption, perhaps we should add a runLog warning when it actually takes place?

Maybe there is a smart way to put a runLog statement in such a way that it doesn't get hit too many times.

Reply:
yep its been default behavior for a while but no reason to not add a warning.

Reply:
single=true is the simple way....